### ISSUE TITLE
Navigation does not work properly when virtual and locked columns are used together.

### ISSUE THREAD

**Issue by gdenchevprog** [telerik/kendo-ui-core#6328]
### Bug report

Reported in 1509795.

### Reproduction of the problem

1. Open and run the following dojo: [https://dojo.telerik.com/OSOCiLal/3](https://dojo.telerik.com/OSOCiLal/3)
2. Use the arrows to navigate the table cells to the right until you reach around columns 13/14/15.
3. Try to navigate back using the arrow keys.

### Current behavior

The navigation does not allow you to go back to the previous columns.

### Expected/desired behavior

The navigation should work in the same manner as it does when only locked or only virtual columns are used.

### Environment

- **Kendo UI version:** 2023.3.1114
- **jQuery version:** 3.4.1
- **Browser:** \[all\]
